File: 1-two-sum/1-two-sum.py
# https://leetcode.com/problems/two-sum/submissions/

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        mapping = {}
        for i, num in enumerate(nums):
            if target-num in mapping:
                return [mapping[target-num], i]
            else:
                mapping[num] = i
        

File: 1-two-sum/1-two-sum.js
// https://leetcode.com/problems/two-sum/

/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
  let obj = {};

  for (let i = 0; i < nums.length; i++) {
    const n = nums[i];
    if (obj[target - n] !== undefined) {
      return [obj[target - n], i];
    } else {
      obj[n] = i;
    }
  }
  return null;
};
